client/src/components/product/product-create-modal.tsx

```diff
@@ -0,0 +1,366 @@
+import React, { useState } from 'react';
+import { toast } from 'react-toastify';
+
+import { FormControl, IconButton, InputLabel, MenuItem, Select, TextField } from '@mui/material';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CircularProgress from '@mui/material/CircularProgress';
+import Modal from '@mui/material/Modal';
+import TextareaAutosize from '@mui/material/TextareaAutosize';
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+
+import { StatusType, useCreateEventMutation } from '../../graphql/graphql-types';
+import { useAuthContext } from '../../store/contexts';
+import { useOrgContext } from 'store/contexts/org.context';
+import { EventStatus, EventTypes } from './constants';
+import { Delete } from '@mui/icons-material';
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  bgcolor: 'background.paper',
+  boxShadow: 24,
+  borderRadius: '7px',
+  p: 4,
+  maxHeight: '90vh',
+  overflow: 'auto',
+};
+
+export interface IProductCreateModal {
+  open: boolean;
+  handleClose: () => void;
+  refetchEvents: () => void;
+}
+
+const ProductCreateModal: React.FC<IProductCreateModal> = ({
+  open,
+  handleClose,
+  refetchEvents,
+}) => {
+  const { user } = useAuthContext();
+  const { org } = useOrgContext();
+
+  const [values, setValues] = useState({
+    name: '',
+    subHeading: '',
+    description: [],
+    startDate: new Date().toISOString(),
+    endDate: new Date().toISOString(),
+    poster: '',
+    prizes: '',
+    contact: [],
+    type: '',
+    status: 'DRAFT',
+  });
+
+  const [createEvent, { error, loading }] = useCreateEventMutation();
+
+  const handleDateTime = (e) => {
+    setValues((current) => ({
+      ...current,
+      date: e.toISOString(),
+    }));
+  };
+
+  const handleChange = (e) => {
+    setValues({
+      ...values,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleArrayChange = (index: number, key: string) => (e) =>
+    setValues((current) => {
+      current[key][index] = e.target.value;
+
+      return {
+        ...current,
+        [key]: current[key],
+      };
+    });
+
+  const handleRemove = (index: number, key: string) => {
+    setValues((current) => {
+      current[key].splice(index, 1);
+      return {
+        ...current,
+        [key]: current[key],
+      };
+    });
+  };
+
+  const handleEventCreate = async () => {
+    try {
+      await createEvent({
```

Comment:
Same here as well. Only this particular organization members with admin access should see the button and be able to add new events

Reply:
It has been managed here https://github.com/dscnitrourkela/project-dates-web/blob/933e7b3db5dd7a140bc2dcbcdbf83716c8865e9a/client/src/components/product/product-list-toolbar.tsx#L34.

Reply:
It has been managed here https://github.com/dscnitrourkela/project-dates-web/blob/933e7b3db5dd7a140bc2dcbcdbf83716c8865e9a/client/src/components/product/product-list-toolbar.tsx#L34-L43